Modules/Core/Common/include/itkDICOMOrientation.h

```diff
@@ -0,0 +1,274 @@
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *         https://www.apache.org/licenses/LICENSE-2.0.txt
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *=========================================================================*/
+#ifndef itkDICOMOrientation_h
+#define itkDICOMOrientation_h
+
+#include "ITKCommonExport.h"
+#include "itkImageBase.h"
+#ifndef ITK_FUTURE_LEGACY_REMOVE
+#include "itkSpatialOrientation.h"
+#endif
+#include <map>
+#include <string>
+
+namespace itk
+{
+
+/** \class DICOMOrientation
+ * \brief A supporting class for DICOMOrientImageFilter.
+ *
+ *  Defines enums to for patient orientation in a compatible way with DICOM.
+ *
+ *  Instances hold the patient orientation enum and allow conversion to and from enums, string and direction cosine
+ * matrices. Conversions from a direction cosine matrix is approximated with the orientation of the closes axes.
+ *
+ * \ingroup ITKCommon
+ */
+class ITKCommon_EXPORT DICOMOrientation
+{
+public:
+  constexpr static unsigned int Dimension = 3;
+  using DirectionType = typename ImageBase<Dimension>::DirectionType;
+  constexpr static unsigned int ImageDimension = Dimension;
+
+#ifndef ITK_FUTURE_LEGACY_REMOVE
+  using LegacyOrientationType = SpatialOrientationEnums::ValidCoordinateOrientations;
+#endif
+
+  enum class CoordinateEnum : uint8_t
+  {
+    UNKNOWN = 0,
+    Right = 2, // 0b0010
+    Left = 3,
+    Anterior = 4,  ///< front - 0b0100
+    Posterior = 5, ///< back
+    Inferior = 8,  ///< bottom - 0b1000
+    Superior = 9   ///< above
+  };
+
+private:
+  enum class CoordinateMajornessTermsEnum : uint8_t
+  {
+    PrimaryMinor = 0,
+    SecondaryMinor = 8,
+    TertiaryMinor = 16
+  };
+
+  constexpr static uint32_t
+  toOrientation(const CoordinateEnum primary, const CoordinateEnum secondary, const CoordinateEnum tertiary)
+  {
+    return (static_cast<uint32_t>(primary) << static_cast<uint8_t>(CoordinateMajornessTermsEnum::PrimaryMinor)) +
+           (static_cast<uint32_t>(secondary) << static_cast<uint8_t>(CoordinateMajornessTermsEnum::SecondaryMinor)) +
+           (static_cast<uint32_t>(tertiary) << static_cast<uint8_t>(CoordinateMajornessTermsEnum::TertiaryMinor));
+  }
+
+  CoordinateEnum
+  GetCoordinateTerm(CoordinateMajornessTermsEnum cmt) const;
+
+public:
+#if 0
+// When this is enabled the following error occurs:
+//ITK/Modules/Core/Common/include/itkDICOMOrientation.h:93:11: error: enumerator value is not a constant expression
+//    RIP = ITK_ORIENTATIONENUM(CoordinateEnum::Right, CoordinateEnum::Inferior, CoordinateEnum::Posterior),
+
```

Comment:
@N-Dekker Any idea why the function does not work for setting enum values?

Reply:
@blowekamp Reproduced as follows: https://godbolt.org/z/hYocrbeb5

```cpp
class MyClass
{
private:
  static constexpr int
  Func42()
  {
    return 42;
  }

  enum class EnumClass
  {
    EnumConst42 = Func42()
  };
};

```

Output from GCC 14.2:
```
<source>:15:25: error: 'static constexpr int MyClass::Func42()' called in a constant expression before its definition is complete
   15 |     EnumConst42 = Func42()
      |                   ~~~~~~^~
<source>:15:26: error: enumerator value for 'EnumConst42' is not an integer constant
   15 |     EnumConst42 = Func42()
```

Reply:
@blowekamp Instead of a static member function, `toOrientation(primary, secondary, tertiary)`, would it be an option for you to add a [variable template](https://en.cppreference.com/w/cpp/language/variable_template) as static member variable to `DICOMOrientation`? For example, `m_ToOrientation`, as follows:

```cpp
template <CoordinateEnum primary, CoordinateEnum secondary, CoordinateEnum tertiary>
static constexpr uint32_t m_ToOrientation =
  (static_cast<uint32_t>(primary) << static_cast<uint8_t>(CoordinateMajornessTermsEnum::PrimaryMinor)) +
  (static_cast<uint32_t>(secondary) << static_cast<uint8_t>(CoordinateMajornessTermsEnum::SecondaryMinor)) +
  (static_cast<uint32_t>(tertiary) << static_cast<uint8_t>(CoordinateMajornessTermsEnum::TertiaryMinor));
```

To be used as follows:

```cpp
enum class OrientationEnum : uint32_t
{
  RIP = m_ToOrientation<CoordinateEnum::LeftToRight, CoordinateEnum::SuperiorToInferior, CoordinateEnum::AnteriorToPosterior>,
  LIP = m_ToOrientation<CoordinateEnum::RightToLeft, CoordinateEnum::SuperiorToInferior, CoordinateEnum::AnteriorToPosterior>,
  ... // etc. etc....
};


```

Reply:
Done

Reply:
This seems to work.